src/Type/Php/ImplodeFunctionReturnTypeExtension.php

```diff
@@ -80,22 +80,23 @@ private function implode(Type $arrayType, Type $separatorType): Type
 		}
 
 		$accessoryTypes = [];
+		$valueTypeAsString = $arrayType->getIterableValueType()->toString();
```

Comment:
the actual change is here: we use `toString` as the `implode` function internally casts everything.

that way the string types detection also works for unions of strings and non-strings and so we get more precise results